### ISSUE TITLE
spack spec/solve include option for --test dependencies

### ISSUE THREAD

**Issue by jacobmerson** [LLNL/spack#23586]
When using the `spack spec` or `spack solve` commands there should be a way to see the test dependencies that are included with `spack install --test  root ...`

### Rationale
It is often helpful to visualize what packages will be built if they are already installed, etc to make sure the packages.yaml file is set up right, or to further constrain things to avoid rebuilding already installed packages. Therefore also being able to see the test dependencies would be helpful.

### Description

Something like `spack spec --install root mypackage` would be a reasonable starting point as it is consistent with `spack install --test root mypackage`

### General information
* **Spack:** 0.16.1-2555-daebf5caf8
* **Python:** 3.8.5
* **Platform:** linux-ubuntu20.04-broadwell
* **Concretizer:** clingo

- [x] I have run `spack --version` and reported the version of Spack
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by vvolkl** [comment]
Agreed! My proposal for a solution is here: https://github.com/spack/spack/pull/24169 (just for `spec` at the moment)

**Comment by alalazo** [comment]
See also #23684 for another PR similar in spirit (but answering a different question).

**Comment by alalazo** [comment]
Related to #29447 in the sense that:
1. Currently we don't have a way to record test dependencies
2. We cannot distinguish between test deps used by `spack test` and test deps used by `spack install --test=*`
